src/docfx/build/redirection/BuildRedirection.cs

```diff
@@ -27,6 +28,13 @@ internal static List<Error> Build(Context context, Document file)
                 file.Mime.Value);
 
             publishItem.RedirectUrl = context.RedirectionProvider.GetRedirectUrl(file.FilePath);
+            var (documentId, documentVersionIndependentId) = context.DocumentProvider.GetDocumentId(context.RedirectionProvider.GetOriginalFile(file.FilePath));
+            publishItem.ExtensionData = new JObject
+            {
+                ["document_id"] = documentId,
+                ["document_version_independent_id"] = documentVersionIndependentId,
+                ["canonical_url"] = file.CanonicalUrl,
```

Comment:
Is `canonical_url` required?  canonical urls are for HTML page SEO purpose, redirection files doesn't get a chance to show the HTML page.

 https://yoast.com/what-is-a-canonical-url/

Reply:
BI team need to know the page url. Provide it from canonical_url or from other metadata both OK to me. But, we need the url.

Reply:
What does v2 and conceptual documents use for this info?